src/modules/SdnDiag.Health/private/Test-NetworkControllerCertCredential.ps1

```diff
@@ -14,20 +14,35 @@ function Test-NetworkControllerCertCredential {
         [System.Management.Automation.Credential()]
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```

Comment:
Review `Mandatory` attribute

Reply:
This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.